src/TargetPlugin.groovy

```diff
@@ -0,0 +1,51 @@
+import static TerraformEnvironmentStage.ALL
+
+class TargetPlugin implements TerraformPlanCommandPlugin, TerraformApplyCommandPlugin, TerraformEnvironmentStagePlugin {
+    public static void init() {
+        TargetPlugin plugin = new TargetPlugin()
+
+        TerraformPlanCommand.addPlugin(plugin)
+        TerraformApplyCommand.addPlugin(plugin)
+        TerraformEnvironmentStage.addPlugin(plugin)
+    }
+
+    @Override
+    public void apply(TerraformPlanCommand command) {
+        def targets = Jenkinsfile.instance.getEnv().RESOURCE_TARGETS
+        if ("${targets}" != '' && targets != null) {
+            for (item in "${targets}".split(',')) {
+                command.withArgument("-target ${item}")
+            }
+        }
+    }
+
+    @Override
+    public void apply(TerraformApplyCommand command) {
+        def targets = Jenkinsfile.instance.getEnv().RESOURCE_TARGETS
+        if ("${targets}" != '' && targets != null) {
+            for (item in "${targets}".split(',')) {
+                command.withArgument("-target ${item}")
+            }
+        }
```

Comment:
This is opinion - you don't have to change the above.  It feels a little wonky.

1.  If condition and logic operators support short-circuiting - once the condition in the left of `&&` evaluates to false, you no longer have to test the right side.  If you're worried about `null`, which you appear to be based on the use of string interpolation (`"${targets}"`) - you have a couple of options.
2.  You can use the elvis operator when you assign `targets` to guarantee that `targets` is a non-null value.  eg: `ef targets = Jenkinsfile.instance.getEnv().RESOURCE_TARGETS ?: ''` https://groovy-lang.org/operators.html#_elvis_operator
3. Once you have a non-null value, `split` doesn't care about empty strings (I bring up empty string, because you explicitly test for it).  Once you can guarantee a non-null value, you can leverage the behavior of `split` to handle the empty string.

Combining all of that, you could have something more concise:

```
        def targets = Jenkinsfile.instance.getEnv().RESOURCE_TARGETS ?: ''
        targets.split(',').each { item -> command.withArgument("-target ${item}") }
```

Reply:
This is a good idea, I'm going to make this change and test it out

Reply:
Tested out and this works perfectly!

Reply:
Looks great!